### PR TITLE
AWS - Contributor Insight Rule resource

### PR DESCRIPTION
**Pull request by Lucas-Irvine**
Add support for the contributor insight rule resource. This is for tagging controls, as well as the ability to disable and delete non-compliant resources.

### CODE REVIEW

File: tests/test_cwi.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2016-2017 Capital One Services, LLC
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from .common import BaseTest
+
+
+class CloudWatchInsightRule(BaseTest):
+
+    def test_disable_insight_rule(self):
```
Comment (PratMis) [inline tests/test_cwi.py:9]: You might want to consider using pytest terraform for testing

**Review by kapilt — COMMENTED** [review]
lgtm, only question is on resource name, insight-rule feels a bit generic but that could just be my own lack of usage.context, we do occasionally do service name prefixes for overly generic capabilities, within the provider namespace, insight-rule seems unlikely to conflict though.

File: tests/test_cwi.py (hunk 1 above)
Comment (PratMis) [reply tests/test_cwi.py:9]: nvm, the above utility is more useful for functional tests

File: c7n/resources/cw.py [hunk 2]
```diff
@@ -166,6 +166,96 @@ def get_arns(self, resources):
...
 
+@resources.register('insight-rule')
+class InsightRule(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'cloudwatch'
+        arn_type = 'insight-rule'
+        enum_spec = ('describe_insight_rules', 'InsightRules', None)
+        name = id = 'Name'
+        universal_taggable = True
```
Comment (PratMis) [inline c7n/resources/cw.py:177]: Should this be changed to `object()` since we don't have to maintain any compatibility

**Comment by Lucas-Irvine** [comment]
> lgtm, only question is on resource name, insight-rule feels a bit generic but that could just be my own lack of usage.context, we do occasionally do service name prefixes for overly generic capabilities, within the provider namespace, insight-rule seems unlikely to conflict though.

Yep, from looking at the AWS docs I think it's mainly referred to as insight-rule or contributor insights rule, and I preferred the former because it seems more concise. I agree it's unlikely to conflict so I think I can leave it as is? Seems similar to the cloudwatch event-rule which also has a name of just event-rule.

File: c7n/resources/cw.py [hunk 3]
```diff
@@ -166,6 +166,96 @@ def get_arns(self, resources):
...
+@resources.register('insight-rule')
+class InsightRule(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'cloudwatch'
+        arn_type = 'insight-rule'
+        enum_spec = ('describe_insight_rules', 'InsightRules', None)
+        name = id = 'Name'
+        universal_taggable = object()
+        permission_augment = ('cloudWatch::ListTagsForResource')
```
Comment (kapilt) [inline c7n/resources/cw.py:178]: this should have a trailing comma

**Comment by kapilt** [comment]
> > lgtm, only question is on resource name, insight-rule feels a bit generic but that could just be my own lack of usage.context, we do occasionally do service name prefixes for overly generic capabilities, within the provider namespace, insight-rule seems unlikely to conflict though.
> 
> Yep, from looking at the AWS docs I think it's mainly referred to as insight-rule or contributor insights rule, and I preferred the former because it seems more concise. I agree it's unlikely to conflict so I think I can leave it as is? Seems similar to the cloudwatch event-rule which also has a name of just event-rule.

let's leave as is

**Review by kapilt — APPROVED** [review]

**Pull request merged**